### PR TITLE
Update `HubAccess` and expose `changesetIndex` in `NamedVersion` entity

### PR DESCRIPTION
**Pull request by aruniverse**

### CODE REVIEW

File: itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts [hunk 1]
```diff
@@ -46,13 +42,28 @@ export class FrontendIModelsAccess implements FrontendHubAccess {
...
+
+    const namedVersionName = version.getName();
+    if (namedVersionName)
+      return this.getChangesetFromNamedVersion({ ...arg, versionName: namedVersionName });
+
+    return this.getLatestChangeset(arg);
   }
 
-  public async getChangesetIdFromNamedVersion(arg: IModelIdArg & { versionName: string }): Promise<ChangesetId> {
+  public async getChangesetFromNamedVersion(arg: IModelIdArg & { versionName?: string }): Promise<ChangesetIndexAndId> {
```
Comment (aruniverse) [inline itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts:66]: As per keith, versionName should be optional, if not provided it should fetch the latest named version. Is this possible with the current api?

Comment (austeja-bentley) [reply itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts:66]: The current API does not support this. As per our discussion, let's postpone this feature to some later time.

Comment (aruniverse) [reply itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts:66]: Sounds good, see https://github.com/iTwin/itwinjs-core/pull/2902

File: itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts [hunk 2]
```diff
@@ -84,4 +98,30 @@ export class FrontendIModelsAccess implements FrontendHubAccess {
       return PlatformToClientAdapter.toAuthorization(token);
     };
   }
+
+  private async _getChangesetFromLatestNamedVersion(arg: IModelIdArg): Promise<ChangesetIndexAndId> {
+    const getNamedVersionListParams: GetNamedVersionListParams = {
```
Comment (austeja-bentley) [inline itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts:103]: could be simplified to:
`const getNamedVersionListParams: GetNamedVersionListParams = this.getIModelScopedOperationParams(arg);`

File: itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts [hunk 3]
```diff
@@ -84,4 +98,30 @@ export class FrontendIModelsAccess implements FrontendHubAccess {
       return PlatformToClientAdapter.toAuthorization(token);
     };
   }
+
+  private async _getChangesetFromLatestNamedVersion(arg: IModelIdArg): Promise<ChangesetIndexAndId> {
```
Comment (austeja-bentley) [inline itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts:102]: same comment about the naming - private methods in this repo are not prefixed with _

**Review by austeja-bentley — CHANGES_REQUESTED** [review]

File: itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts [hunk 4]
```diff
@@ -62,9 +76,9 @@ export class FrontendIModelsAccess implements FrontendHubAccess {
 
     const namedVersionsIterator: AsyncIterableIterator<MinimalNamedVersion> = this._iModelsClient.namedVersions.getMinimalList(getNamedVersionListParams);
     const namedVersions: MinimalNamedVersion[] = await take(namedVersionsIterator, 1);
-    if (namedVersions.length === 0 || !namedVersions[0].changesetId)
+    if (namedVersions.length === 0|| !namedVersions[0].changesetId)
```
Comment (austeja-bentley) [inline itwin-platform-access/imodels-access-frontend/src/FrontendIModelsAccess.ts:79]: nitpick: a space is missing :)

**Review by austeja-bentley — APPROVED** [review]

**Pull request merged**